smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt

```diff
@@ -4,111 +4,72 @@
  */
 package software.amazon.smithy.swift.codegen
 
-import software.amazon.smithy.build.FileManifest
 import software.amazon.smithy.codegen.core.SymbolDependency
-import software.amazon.smithy.swift.codegen.resources.Resources
-import software.amazon.smithy.utils.CodeWriter
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
 
-fun writePackageManifest(settings: SwiftSettings, fileManifest: FileManifest, dependencies: List<SymbolDependency>, generateTestTarget: Boolean = false) {
+class PackageManifestGenerator(val ctx: ProtocolGenerator.GenerationContext) {
 
-    // filter duplicates in dependencies
-    val distinctDependencies = dependencies.distinctBy { it.packageName }
-    val writer = CodeWriter().apply {
-        trimBlankLines()
-        trimTrailingSpaces()
-        setIndentText("    ")
-    }
+    fun writePackageManifest(dependencies: List<SymbolDependency>) {
+        ctx.delegator.useFileWriter("Package.swift") { writer ->
+            writer.write("// swift-tools-version:\$L", ctx.settings.swiftVersion)
+            writer.write("")
+            writer.write("import PackageDescription")
+            writer.write("")
```

Comment:
`Package.swift` is an exception among generated files: it writes its own import statements, since it needs to have special content at the top of the file.  Going forward, though, the only import in `Package.swift` should be `PackageDescription`.